src/lib.rs

```diff
@@ -226,6 +234,62 @@ impl Response {
     }
 }
 
+async fn multipart_get(slice: &'static mut [u8], path: &Path, client: &dyn ObjectStore) -> anyhow::Result<usize, Box<dyn Error>> {
+    let part_size: usize = static_config().multipart_get_part_size as usize;
+    let result = client.head(&path).await?;
+    if result.size > slice.len() {
+        return Err("Supplied buffer was too small".into());
+    }
+
+    // If the object size happens to be smaller than part_size,
+    // then we will end up doing a single range get of the whole
+    // object.
+    let mut parts = result.size / part_size;
+    if result.size % part_size != 0 {
+        parts += 1;
+    }
+    let mut part_ranges = Vec::with_capacity(parts);
+    for i in 0..(parts-1) {
+        part_ranges.push((i*part_size)..((i+1)*part_size));
+    }
+    // Last part which handles sizes not divisible by part_size
+    part_ranges.push(((parts-1)*part_size)..result.size);
+
+    let result_vec = client.get_ranges(&path, &part_ranges).await?;
+    let accum = tokio::spawn(async move {
+        let mut accum: usize = 0;
+        for i in 0..result_vec.len() {
+            slice[accum..accum + result_vec[i].len()].copy_from_slice(&result_vec[i]);
+            accum += result_vec[i].len();
+        }
+        accum
+    }).await?;
+
+    return Ok(accum);
+}
+
+async fn multipart_put(slice: &'static [u8], path: &Path, client: &dyn ObjectStore) -> anyhow::Result<(), Box<dyn Error>> {
+    let (multipart_id, mut writer) = client.put_multipart(&path).await?;
+    match writer.write_all(slice).await {
+        Ok(_) => {
+            match writer.flush().await {
+                Ok(_) => {
+                    writer.shutdown().await?;
+                    return Ok(());
+                }
+                Err(e) => {
```

Comment:
err =>
...
err?

Reply:
I wasn't able to get this to work, it seemed to get confused on the return type of the function. Using e.into instead worked.